comment/mixins.py

```diff
@@ -0,0 +1,17 @@
+from django.core.exceptions import PermissionDenied
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.shortcuts import get_object_or_404
+
+from comment.models import Comment
+
+
+class AnonymousPermissionMixin:
+    def has_permission(self, request, view):
+        self.comment = get_object_or_404(Comment, pk=self.kwargs.get('pk'))
+        if not self.comment.user:
+            raise PermissionDenied
```

Comment:
I believe this is doing same what `LoginRequiredMixin` is doing.

Reply:
`LoginRequiredMixin` checks whether the user is logged in or not. This checks whether the comment has an author or not.

Reply:
Yes. what I meant here that the chack you mentioned is already implemented in dispatch method and `AnonymousPermissionMixin` no longer needed

Reply:
Yes, I agree.